heareval/predictions/task_predictions.py

```diff
@@ -749,30 +779,153 @@ def dataloader_from_split_name(
 class GridPointResult:
     def __init__(
         self,
-        model_path: str,
         epoch: int,
-        time_in_min: float,
         hparams: Dict[str, Any],
         postprocessing: Tuple[Tuple[str, Any], ...],
-        trainer: pl.Trainer,
         validation_score: float,
         score_mode: str,
+        conf: Dict,
     ):
-        self.model_path = model_path
         self.epoch = epoch
-        self.time_in_min = time_in_min
         self.hparams = hparams
         self.postprocessing = postprocessing
-        self.trainer = trainer
         self.validation_score = validation_score
         self.score_mode = score_mode
+        self.conf = conf
 
 
-def task_predictions_train(
+def task_predictions_train_test_full(
+    best_grid_point: GridPointResult,
     embedding_path: Path,
     embedding_size: int,
-    grid_points: int,
     metadata: Dict[str, Any],
+    data_splits: Dict[str, List[str]],
```

Comment:
This adds a lot of code but it's not clear what's the difference between this and below. That makes it quite confusing and I'm worried about the code duplication

Reply:
Agreed - I'm not really sure we need two different train functions. I understand the idea was to train the final model using the train + val to maximize the data we're using. But I think we still want to use the validation when training the final model and use early stopping.